p2p/peer.py

```diff
@@ -238,11 +239,12 @@ def remove_subscriber(self, subscriber: 'asyncio.Queue[PEER_MSG_TYPE]') -> None:
         if subscriber in self._subscribers:
             self._subscribers.remove(subscriber)
 
-    async def do_sub_proto_handshake(self):
+    async def do_sub_proto_handshake(self) -> None:
         """Perform the handshake for the sub-protocol agreed with the remote peer.
 
         Raises HandshakeFailure if the handshake is not successful.
         """
+        self.base_protocol = cast(P2PProtocol, self.base_protocol)
```

Comment:
`BasePeer.__init__()` does `self.base_protocol = P2PProtocol(self)`, so I'm wondering why this is needed

Reply:
Yep, we can do better here. Checkout https://github.com/ethereum/py-evm/pull/947/commits/a8632048824259f6816b57aca92c7c72d8c36d24

Basically we:

- get rid of the type of `self.base_protocol` on the class level so that it is really only defined by `__init__`
- return early in `get_protocol_command_for` so that the intermediate variable doesn't have to be types as `Union[Protocol, P2PProtocol]`